Converters/Revit/Speckle.Converters.RevitShared/ToSpeckle/Raw/Geometry/MeshByMaterialDictionaryToSpeckle.cs

```diff
@@ -34,15 +34,25 @@ public MeshByMaterialDictionaryToSpeckle(
   /// These meshes are created with an initial capacity based on the size of the vertex and face arrays to avoid unnecessary resizing.
   /// Also note that, for each unique material, the method tries to retrieve the related DB.Material from the current document and convert it. If the conversion is successful,
   /// the material is added to the corresponding Speckle mesh. If the conversion fails, the operation simply continues without the material.
+  /// TODO: update description
   /// </remarks>
-  public List<SOG.Mesh> Convert(Dictionary<DB.ElementId, List<DB.Mesh>> target)
+  public List<SOG.Mesh> Convert((Dictionary<DB.ElementId, List<DB.Mesh>> target, DB.ElementId parentElementId) args)
   {
-    var result = new List<SOG.Mesh>(target.Keys.Count);
+    var result = new List<SOG.Mesh>(args.target.Keys.Count);
+    var objectRenderMaterialProxiesMap = _contextStack.RenderMaterialProxyCache.ObjectRenderMaterialProxiesMap;
 
-    foreach (var meshData in target)
+    var materialProxyMap = new Dictionary<string, RenderMaterialProxy>();
+    objectRenderMaterialProxiesMap[args.parentElementId.ToString()!] = materialProxyMap;
```

Comment:
how confident we are, we always reset the list of material proxies of `parent` in the cache?